data/types/device/alarm/alarm.go

```diff
@@ -120,14 +123,23 @@ func (a *Alarm) Validate(validator structure.Validator) {
 		}
 		validator.String("statusId", a.StatusID).Using(data.IDValidator)
 	}
-	validator.String("ackStatus", a.AckStatus).OneOf(AckStatuses()...)
-	if a.EventID != nil {
-		validator.String("alarmLevel", a.AlarmLevel).Exists().OneOf(AlarmLevels()...)
-		validator.String("alarmCode", a.AlarmCode).Exists()
-		validator.String("alarmLabel", a.AlarmLabel).Exists()
-		validator.String("ackStatus", a.AckStatus).Exists()
-		timeValidator := validator.String("updateTime", a.UpdateTime)
-		timeValidator.AsTime(types.TimeFormat).Exists()
+	if a.AlarmType != nil {
+		if *a.AlarmType == AlarmTypeHandset {
+			validator.String("eventID", a.EventID).Exists()
+			validator.String("alarmLevel", a.AlarmLevel).Exists().OneOf(AlarmLevels()...)
+			validator.String("alarmCode", a.AlarmCode).Exists()
+			validator.String("alarmLabel", a.AlarmLabel).Exists()
+			validator.String("ackStatus", a.AckStatus).Exists()
+			timeValidator := validator.String("updateTime", a.UpdateTime)
+			timeValidator.AsTime(types.TimeFormat).Exists()
+		} else {
+			validator.String("ackStatus", a.AckStatus).OneOf(AckStatuses()...)
+			validator.String("alarmCode", a.AlarmCode)
+			validator.String("alarmLabel", a.AlarmLabel)
+			validator.String("ackStatus", a.AckStatus)
+			timeValidator := validator.String("updateTime", a.UpdateTime)
+			timeValidator.AsTime(types.TimeFormat)
+		}
```

Comment:
alarmLevel is not always validated with the list of Alarm Levels
```suggestion

	alarmLevelValidator := validator.String("alarmLevel", a.AlarmLevel)
	alarmLevelValidator.OneOf(AlarmLevels()...)
	
	ackStatusValidator := validator.String("ackStatus", a.AckStatus)
	ackStatusValidator.OneOf(AckStatuses()...)
	
	timeValidator := validator.String("updateTime", a.UpdateTime)
	timeValidator.AsTime(types.TimeFormat)
	
	if a.AlarmType !=nil && *a.AlarmType == AlarmTypeHandset {
		alarmLevelValidator.Exists()
		ackStatusValidator.Exists()
		validator.String("eventID", a.EventID).Exists()
		validator.String("alarmCode", a.AlarmCode).Exists()
		validator.String("alarmLabel", a.AlarmLabel).Exists()
		timeValidator.Exists()
	}
```